controllers/crunchybridge/bridgecluster_controller.go

```diff
@@ -39,17 +53,117 @@ type BridgeClusterReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the BridgeCluster object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *BridgeClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx, "BridgeCluster", req.NamespacedName)
+
+	if r.BridgeClient == nil {
+		err := errors.New("Uninitialized client")
+		logger.Error(err, "No CrunchyBridge client configured")
+		return ctrl.Result{}, err
+	}
+
+	clusterObj := &crunchybridgev1alpha1.BridgeCluster{}
+	if err := r.Get(ctx, req.NamespacedName, clusterObj); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Likely deleted before action or extra pass post-deletion, no-op
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "Error fetching BridgeCluster object for reconciliation")
+		return ctrl.Result{}, err
+	}
+
+	if clusterObj.DeletionTimestamp != nil && !clusterObj.DeletionTimestamp.IsZero() {
+		// Cluster deletion request / process finalizer
+		if listContains(clusterObj.Finalizers, bcFinalizer) {
+			if id := clusterObj.Status.Cluster.ID; id != "" {
+				logger.Info("deleting cluster", "id", id)
+				if err := r.BridgeClient.DeleteCluster(id); err != nil {
+					return ctrl.Result{}, err
+				}
+			}
+			controllerutil.RemoveFinalizer(clusterObj, bcFinalizer)
+			if err := r.Update(ctx, clusterObj); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		switch clusterObj.Status.Phase {
+		case crunchybridgev1alpha1.PhaseUnknown:
```

Comment:
Switch/case format in the reconcile's a nice touch - I feel a nice visual break between the responsibilities and easily identifiable parts for future reference :clap: